frontend/sw360-portlet/src/main/webapp/js/modules/sw360Wizard.js

```diff
@@ -0,0 +1,244 @@
+     * displaying user feedback before returning.
+     * If the request fails for technical reasons, you can handle that case in the submitErrorHook of the
+     * still current step.
+     * But if the request succeeded and you just do not receive the expected data from the backend, you have
+     * to advise the user what he needs to do in the next renderHook as the step will be switched no matter
+     * what you are trying to return from your renderHook.
+     *
+     * When a user has clicked through all steps, the next button caption is changed into finish and the
+     * data received from this backend call will be passed into the special finishCb where you might want to
+     * redirect your user onto another page.
+     */
+    var sw360Wizard = function(config) {
+        var $wizardRoot = $(config.wizardRoot),
+            firstStep = $('.wizardBody div.step:first', $wizardRoot),
+            lastStep = $('.wizardBody div.step:last', $wizardRoot);
+
+        $('head').append('<link rel="stylesheet" href="/sw360-portlet/css/sw360Wizard.css" type="text/css" />');
+
+        $wizardRoot.append('' +
+            '<div class="wizardFooter">' +
+            '    <input type="button" class="wizardNext addButton" value="Next" />' +
+            '    <input type="button" class="wizardBack cancelButton" value="Back" disabled/>' +
+            '</div>'
+        );
+
+        function changeHeaderState(lastActiveIndex, activeIndex) {
+            $('.wizardHeader li', $wizardRoot).each(function(index, element) {
+                if (index === lastActiveIndex) {
+                    $(element).removeClass('active');
+                }
+                if (index === activeIndex) {
+                    $(element).addClass('active');
+                }
+            });
+        }
+
+        function determineAndSetFooterState() {
+            if (firstStep.hasClass('active')) {
+                $('.wizardBack', $wizardRoot).attr('disabled', true);
+            } else {
+                $('.wizardBack', $wizardRoot).attr('disabled', false);
+            }
+
+            if (lastStep.hasClass('active')) {
+                $('.wizardNext', $wizardRoot).val('Finish')
+            } else {
+                $('.wizardNext', $wizardRoot).val('Next');
+            }
+        }
+
+        function reworkPostData(data) {
+            var reworkedPostData = {};
+            $.each(data, function(key, value) {
+                if (typeof value === 'object') {
+                    value = JSON.stringify(value);
+                }
+                reworkedPostData[config.postParamsPrefix + key] = value;
+            });
+
+            return reworkedPostData;
+        }
+
+        function fillContentOfStep(activeElement, activeIndex, nextElement, nextIndex) {
+            var proceed = true,
+                proceedTmp;
+
+            if (nextElement[0] === firstStep[0]) {
```

Comment:
I'd like to see this unboxing of single-element arrays happen right where the search for them is performed with `$(...)`. It looks a little odd that variables named in singular (e.g. "nextElemen**t**") happen to be arrays.